pkg/api/admin/api/config.go

```diff
@@ -0,0 +1,142 @@
+	MasterEtcd            *string `json:"masterEtcd,omitempty"`
+	ControlPlane          *string `json:"controlPlane,omitempty"`
+	Node                  *string `json:"node,omitempty"`
+	ServiceCatalog        *string `json:"serviceCatalog,omitempty"`
+	Sync                  *string `json:"sync,omitempty"`
+	TemplateServiceBroker *string `json:"templateServiceBroker,omitempty"`
+	Registry              *string `json:"registry,omitempty"`
+	Router                *string `json:"router,omitempty"`
+	RegistryConsole       *string `json:"registryConsole,omitempty"`
+	AnsibleServiceBroker  *string `json:"ansibleServiceBroker,omitempty"`
+	WebConsole            *string `json:"webConsole,omitempty"`
+	Console               *string `json:"console,omitempty"`
+	EtcdBackup            *string `json:"etcdBackup,omitempty"`
+
+	// GenevaImagePullSecret defines secret used to pull private Azure images
+	GenevaImagePullSecret []byte `json:"genevaImagePullSecret,omitempty"`
+	// Geneva integration images
+	GenevaLogging *string `json:"genevaLogging,omitempty"`
+	GenevaTDAgent *string `json:"genevaTDAgent,omitempty"`
+}
+
+// CertificateConfig contains all certificate configuration for the cluster.
+type CertificateConfig struct {
+	// CAs
+	EtcdCa           *CertKeyPair `json:"etcdCa,omitempty"`
+	Ca               *CertKeyPair `json:"ca,omitempty"`
+	FrontProxyCa     *CertKeyPair `json:"frontProxyCa,omitempty"`
+	ServiceSigningCa *CertKeyPair `json:"serviceSigningCa,omitempty"`
+	ServiceCatalogCa *CertKeyPair `json:"serviceCatalogCa,omitempty"`
+
+	// etcd certificates
+	EtcdServer *CertKeyPair `json:"etcdServer,omitempty"`
+	EtcdPeer   *CertKeyPair `json:"etcdPeer,omitempty"`
+	EtcdClient *CertKeyPair `json:"etcdClient,omitempty"`
+
+	// control plane certificates
+	MasterServer         *CertKeyPair `json:"masterServer,omitempty"`
+	OpenshiftConsole     *CertKeyPair `json:"openshiftConsole,omitempty"`
+	Admin                *CertKeyPair `json:"admin,omitempty"`
+	AggregatorFrontProxy *CertKeyPair `json:"aggregatorFrontProxy,omitempty"`
+	MasterKubeletClient  *CertKeyPair `json:"masterKubeletClient,omitempty"`
+	MasterProxyClient    *CertKeyPair `json:"masterProxyClient,omitempty"`
+	OpenShiftMaster      *CertKeyPair `json:"openShiftMaster,omitempty"`
+	NodeBootstrap        *CertKeyPair `json:"nodeBootstrap,omitempty"`
+
+	// infra certificates
+	Registry                *CertKeyPair `json:"registry,omitempty"`
+	Router                  *CertKeyPair `json:"router,omitempty"`
+	ServiceCatalogServer    *CertKeyPair `json:"serviceCatalogServer,omitempty"`
+	ServiceCatalogAPIClient *CertKeyPair `json:"serviceCatalogAPIClient,omitempty"`
+
+	// misc certificates
+	AzureClusterReader *CertKeyPair `json:"azureClusterReader,omitempty"`
+
+	// geneva integration certificates
+	GenevaLogging *CertKeyPair `json:"genevaLogging,omitempty"`
+}
+
+// CertKeyPair is an rsa private key and x509 certificate pair.
+type CertKeyPair struct {
+	Key  *rsa.PrivateKey   `json:"key,omitempty"`
```

Comment:
Think I'd take the approach of keeping the type, removing the `Key` field, and possibly renaming the type (can't think of a great name though).  That will keep the actual JSON structure as similar as possible.